### ISSUE TITLE
All commands should take allow option of path to preferences directory.

### ISSUE THREAD

**Issue by robbat2** [gilbertchen/duplicacy#148]
If backing up an entire system, having to have the `.duplicacy` directories or files scattered everywhere that you are backing up is very messy, and some of the locations might not permit it (eg backing up all of /home when each homedir is mounted by autofs dynamically, and writing to /home itself is disallowed).

ALL of the commands should support a command-line option to the specific `.duplicacy` file or directory to use (and should make it ignoring the existence of any `.duplicacy` files locally).

**Comment by jbrodriguez** [comment]
This is something I mentioned [here](https://github.com/gilbertchen/duplicacy/issues/132#issuecomment-325848185) too, and @gilbertchen replied he would add --includes/--excludes from the command line

**Comment by tbain98** [comment]
+1

I'm intending to use Duplicacy as only one of the tools performing backups on my data (I don't want to be reliant on only one piece of software, any more than I want to be reliant on any one storage provider), and I hate the idea that each of them will force me to put their "droppings" within the content to be backed up. Being able to externalize that configuration to another location sounds like a positive improvement to me.

**Comment by tbain98** [comment]
@jbrodriguez What @robbat2 is proposing here is a little different from what you're advocating for in that thread. I think there's value to having a directory that contains configuration settings for the content to be backed up (whereas you appeared to be advocating a settings-free approach), but I'm 100% behind @robbat2's suggestion that the settings directory not having to be within the content to be backed up. This would also facilitate the use case you were asking for where you've got multiple backup sets defined from different subsets of the contents of a given folder, which Duplicacy currently can't support because there can be only one `.duplicacy` file for a given root folder.

**Comment by jbrodriguez** [comment]
@tbain98, yes I like the idea of specifying a ```.duplicacy``` from the command line, but you should be able to set filters to add folders from anywhere in the file system (any volume, any subfolder) to your 'backup set'.

I'm just not so sure that this is so explicit (possible?) right now.

**Comment by jsreynolds** [comment]
Am I missing something, or can't you just use `--pref-dir` during the init and place all of the backup configuration settings somewhere else entirely, and/or back them up separately if so desired? You can even move them later rather easily too.

**Comment by robbat2** [comment]
@jsreynolds If you pass `--pref-dir`, it still drops a `.duplicacy` file, containing the path to  the `pref-dir`. I don't want that file to exist at all (and instead I will pass the path to the `pref-dir` via CLI arg and/or environment).

One of the things, that illustrates my goals and @jbrodriguez's request, is that I believe all backups should be taken relative to the root(s) of the entire filesystem. `/` for *nix stuff, and `C:\, D:\, ...` for windows (and the drive letter could be valuable in the backup). That said, I want different retention rules for `/etc` than `/home` (this covers include/excludes too), which means separate `pref-dir`.

The other thing that will need work is that seperating out the cache directory from `pref-dir`, because while I would want all of the `pref-dir`'s to live under something like `/etc/duplicacy/$FOO`, I really don't want cache content to live in `/etc` at all! Should be trivial to add a `cache-dir` param to the preferences.

**Comment by tbain98** [comment]
@jbrodriguez I agree, it should be possible to specify multiple root folders (which need not have an included common parent) for a given backup set, and I agree that it is not currently something that's supported. I want that as well, but this particular issue is about not forcing `.duplicacy` to live within the folder being backed up.

**Comment by tbain98** [comment]
@robbat2 I'm hesitant about what you're proposing with only using absolute paths. In particular, I'd be very reluctant to say that it shouldn't be possible to restore content into a different absolute path than the one they were in when backed up, because I think that greatly restricts your ability to perform restores (and therefore to test whether your backup works properly).

I'd prefer to see the paths for a given root folder stored as absolute paths in the preferences file (so everything still works if you move that file around), but to store only relative paths within the backup itself. Then when you init a repository with the same ID somewhere else, you could specify a different absolute path for the same folder, and a restore to that repository will write the content to that other location, using the relative paths stored within the backed-up content as applied to the absolute path of the root folder.

**Comment by robbat2** [comment]
@tbain98 i certainly would want to restore to a different path, but I want the backup to be taken from the absolute root.  Say I take backups of `/etc` and `/var/lib`, both relative to `/`, then I restore both of them to `/mnt/verify/`, it should contain `/mnt/verify/etc` AND `/mnt/verify/var/lib`

**Comment by tbain98** [comment]
I understand, and I'm not proposing that it should be impossible to do what you're proposing, but I'd like to have the ability to restore those directories as siblings to one another if I wanted to. Put another way, I want to include the folders "etc" and "lib" in the backup, but not include the folder "var" in any way (other than maybe seeding the preferences file with it when I init a new repo). What I'm proposing allows the behavior you're proposing (though it might require slightly more effort to set up a second repository in another location), but also gives additional flexibility.
